pkg/codec/config.go

```diff
@@ -250,3 +268,81 @@ func (h *modifierMarshaller[T]) MarshalJSON() ([]byte, error) {
 
 	return json.Marshal(m)
 }
+
+func decode(bts []byte, val any) error {
+	decoder := json.NewDecoder(bytes.NewBuffer(bts))
+	decoder.UseNumber()
+
+	return decoder.Decode(val)
+}
+
+type Number string
+
+func (n Number) Float64() (float64, error) {
+	return json.Number(n).Float64()
+}
+
+func (n Number) Int64() (int64, error) {
+	return json.Number(n).Int64()
+}
+
+func (n Number) MarshalCBOR() ([]byte, error) {
+	if strings.Contains(string(n), ".") {
+		// parse as float64 and encode
+		floatVal, err := strconv.ParseFloat(string(n), 64)
+		if err != nil {
+			return nil, err
+		}
+
+		return cbor.Marshal(floatVal)
+	}
+
+	// parse as int64 and encode
+	intVal, err := strconv.ParseInt(string(n), 10, 64)
+	if err != nil {
+		return nil, err
+	}
+
+	return cbor.Marshal(intVal)
+}
+
+func (n *Number) UnmarshalCBOR(data []byte) error {
+	fmt.Println("unmarshalling Number")
+
+	var value string
+
+	if err := cbor.Unmarshal(data, &value); err != nil {
+		return err
+	}
+
+	*n = Number(value)
+
+	return nil
+}
+
+func (n Number) MarshalJSON() ([]byte, error) {
+	buffer := bytes.NewBuffer([]byte{})
+	encoder := json.NewEncoder(buffer)
+
+	if err := encoder.Encode(json.Number(n)); err != nil {
+		return nil, err
+	}
+
+	return buffer.Bytes(), nil
+}
+
+func (n *Number) UnmarshalJSON(data []byte) error {
+	fmt.Println("unmarshalling Number as JSON")
+	decoder := json.NewDecoder(bytes.NewBuffer(data))
+	decoder.UseNumber()
+
+	var value json.Number
+
+	if err := decoder.Decode(&value); err != nil {
+		return err
+	}
+
+	*n = Number(value)
+
+	return nil
```

Comment:
can't this just be 

```
var value json.Number
err := json.Unmarshal(data, &value)
*n = Number(value)
return err
```

Reply:
I wasn't sure if I needed the `UseNumber` bit or not. I'll try it without and see how it goes.